jenkins/bokchoy_db_pull_request.py

```diff
@@ -0,0 +1,212 @@
+            "Check to make sure this branch doesn't already exist.".format(branch_name)
+        )
+
+
+def _update_file(repository, file_path, commit_message, contents, file_sha, branch_name):
+    """
+    Create a commit on a branch that updates the file_path with the string contents.
+    """
+    try:
+        repository.update_file(file_path, commit_message, contents, file_sha, branch_name)
+    except:
+        raise StandardError(
+            "Error updating database file: {}".format(file_path)
+        )
+
+
+def _create_pull_request(repository, title, body, base, head):
+    """
+    Create a new pull request with the changes in head.
+    """
+    try:
+        pull_request = repository.create_pull(
+            title=title,
+            body=body,
+            base=base,
+            head=head
+        )
+    except:
+        raise StandardError(
+            "Could not create pull request"
+        )
+
+
+@click.command()
+@click.option(
+    '--sha',
+    help="Sha of the merge commit to base the new PR off of",
+    required=True,
+)
+@click.option(
+    '--github_token',
+    help="Github token for authentication",
+    required=True,
+)
+@click.option(
+    '--repo_root',
+    help="Path to local edx-platform repository that will "
+         "hold updated database files",
+    required=True,
+)
+def main(sha, github_token, repo_root):
+    # Connect to github
+    logger.info("Authenticating with Github")
+    github_instance = _authenticate_with_github(github_token)
+    repository = _connect_to_repo(github_instance, "edx-platform")
+
+    # Iterate through the db files and update them accordingly
+    branch_created = False
+    for db_file in BOKCHOY_DB_FILES:
+        repo_file_path = os.path.join('/', DB_CACHE_FILEPATH, db_file)
+
+        # Get the sha of the file in the repository
+        file_sha = _get_file_sha(repository, repo_file_path)
+
+        # Read the local db files that were updated by paver
+        local_file_path = os.path.join(repo_root, DB_CACHE_FILEPATH, db_file)
+        local_file_data = _read_file_contents(local_file_path)
+
+        # Check if the local file is different from whats in the repo currently
+        if _file_has_changed(repository, local_file_data, repo_file_path):
+            logger.info("File {} has changed.".format(repo_file_path))
+            # Since there are changes needed, create a new branch if we haven't already
+            if not branch_created:
+                # Create a new branch for the db file changes
+                branch_name = "refs/heads/bokchoy_auto_cache_update_" + sha
+                _create_branch(repository, branch_name, sha)
+                branch_created = True
+
+            # Update the db files on our branch to reflect the new changes
+            logger.info("Updating database file: {}".format(repo_file_path))
+            _update_file(repository, repo_file_path, 'Updating migrations', local_file_data, file_sha, branch_name)
```

Comment:
If I am following this correctly, this will create a commit for each updated file in the PR. We would need to manually squash each PR then re-push and merge. Perhaps there is another way add the modified files in one action/function.

Reply:
I'll glance a bit more for this but I haven't found any way around it. We could just merge the prs with the "Squash and merge" button. But obviously it would be ideal to start with just one commit